karate-core/src/main/java/com/intuit/karate/debug/DapServerHandler.java

```diff
@@ -404,11 +406,32 @@ private void start() {
         if (runnerThread != null) {
             runnerThread.interrupt();
         }
+
+        // html report is enabled by default
+        boolean outputHtmlReport = options.getFormats() == null || (options.getFormats() != null && !options.getFormats().contains("~html"));
+        boolean outputCucumberJson = options.getFormats() != null && options.getFormats().contains("cucumber:json");
+        boolean outputJunitXml = options.getFormats() != null && options.getFormats().contains("junit:xml");
+
+        // for the debugger, if the output dir is the default
+        // create a subfolder so multiple executions don't override the previous reports
+        String outputDir = options.getOutput();
+        if(outputDir.contentEquals(FileUtils.getBuildDir())) {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyyMMdd_HHmmss");
```

Comment:
I may refactor this to reuse Suite.backup...() https://github.com/intuit/karate/blob/develop/karate-core/src/main/java/com/intuit/karate/Suite.java#L324 - let me know if ok

Reply:
Yup all good - I searched by date formatter usages and couldn't spot any which is why added it. I think the important part here is being a unique folder for local debugging - different use case than a CI/CD for example where you always want in the same folder for your Jenkins/etc to pick it up from there.